microhapulator/api.py

```diff
@@ -557,3 +559,36 @@ def type(bamfile, markertsv, minbasequal=10, max_depth=1e6):
         for haplotype, count in htcounts.items():
             result.record_haplotype(locusid, haplotype, count)
     return result
+
+
+def read_length_dist(fastq, outfile, scale=1000, title=None):
+    """Plot distribution of read lengths
+
+    :param str fastq: path of a FASTQ file containing NGS reads
+    :param str outfile: path of a graphic file to create
+    :param float scale: scaling factor for the Y axis
+    :param str title: title for the plot
+    """
+    backend = matplotlib.get_backend()
+    plt.switch_backend("Agg")
```

Comment:
It throws gnarly warning messages unless you set the PyPlot backend like so. I reset it at the end of the function in hopes that it doesn't cause issues e.g. when running things in a Jupyter notebook.